torch/cuda/amp/grad_scaler.py

```diff
@@ -329,8 +328,7 @@ def step(self, optimizer, *args, **kwargs):
 
         assert len(optimizer_state["found_inf_per_device"]) > 0, "No inf checks were recorded for this optimizer."
 
-        if not sum(v.item() for v in optimizer_state["found_inf_per_device"].values()):
-            retval = optimizer.step(*args, **kwargs)
+        retval = self._maybe_opt_step(optimizer, optimizer_state, *args, **kwargs)
```

Comment:
Why did you refactor this into a separate function?   Seems to me it's a no-op change that takes 6 lines instead of 2 lines, I don't understand the motivation.

Reply:
For minimizing the code change in torch/xla, [see](https://github.com/pytorch/xla/pull/2654/files#diff-7a0cc30ed2a162ddd69618394c5b4ed5c84d046189eb8c6879b9b38df45b6f3eR5).